### PR TITLE
Add Map() and FilterFunc()

### PR DESCRIPTION
**Pull request by AnirudhVyas**
Please review again and merge.

### CODE REVIEW

File: properties.go [hunk 1]
```diff
@@ -647,6 +622,33 @@ func (p *Properties) WriteComment(w io.Writer, prefix string, enc Encoding) (n i
 	return
 }
 
+// ToMap returns a copy of the properties as a map - useful when users of library are themselves
```
Comment (magiconair) [inline properties.go]: Pls remove the ` - useful ...` part since you don't know what users will want to use this for.

File: properties.go [hunk 2]
```diff
@@ -647,6 +622,33 @@ func (p *Properties) WriteComment(w io.Writer, prefix string, enc Encoding) (n i
...
+// E.g. `Properties` as a return type in library which is using magiconair Properties is not suitable.
+func (p *Properties) ToMap() map[string]string {
+	m := make(map[string]string)
+	for k, v := range p.m {
+		m[k] = v
+	}
+	return m
+}
+
+/// FilterFunc returns a copy of the properties which includes the values
```
Comment (magiconair) [inline properties.go]: Only two slashes for the comment.

File: properties_test.go [hunk 3]
```diff
@@ -825,43 +825,29 @@ func TestMerge(t *testing.T) {
...
-	assert.Equal(t, input.ToMap(), otherMap)
+func TestToMap(t *testing.T) {
+	input := "key=value\nabc=def"
+	p := mustParse(t, input)
+	m := map[string]string{"key": "value", "abc": "def"}
+	assert.Equal(t, p.ToMap(), m)
 }
 
-func TestProperties_ToMapWith(t *testing.T) {
+func TestFilterFunc(t *testing.T) {
```
Comment (magiconair) [inline properties_test.go]:
```go
func TestFilterFunc(t *testing.T) {
    input := "key=value\nabc=def"
    p := mustParse(t, input)
    pp := p.FilterFunc(func(k, v string) bool {
        return k != "abc"
    })
    m := map[string]string{"key": "value"}
    assert.Equal(t, pp.ToMap(), m)
}
```

File: properties.go [hunk 4]
```diff
@@ -622,6 +622,30 @@ func (p *Properties) WriteComment(w io.Writer, prefix string, enc Encoding) (n i
 	return
 }
 
+// ToMap returns a copy of the properties as a map
```
Comment (magiconair) [inline properties.go]: Pls add a full stop at the end.

File: properties_test.go [hunk 5]
```diff
@@ -825,6 +825,22 @@ func TestMerge(t *testing.T) {
 	assert.Equal(t, p1.MustGet("key"), "another value")
 	assert.Equal(t, p1.GetComment("key"), "another comment")
 }
+func TestToMap(t *testing.T) {
```
Comment (magiconair) [inline properties_test.go]: Pls add a blank line before.

**Comment by AnirudhVyas** [comment]
Done

**Comment by AnirudhVyas** [comment]
blank line - added.

File: properties.go (hunk 4 above)
Comment (AnirudhVyas) [inline properties.go]: name it as Map() -- it is mirror image of String( ) ?

Comment (AnirudhVyas) [inline properties.go]: added full stop and pushed ...

Comment (AnirudhVyas) [inline properties.go]: Let me know -- I can rename and push - thanks.

Comment (magiconair) [inline properties.go]: Maybe you can do a quick dig through the stdlib to find some precedence. I don't think they have `ToXXX` functions there. `Map` could work but might be too generic.

**Comment by AnirudhVyas** [comment]
Please let me know if there's anything else that is needed; You may choose a better name instead of ToMap( ). Thanks for your help!

**Comment by magiconair** [comment]
The code looks good. Kids kept me busy today so I didn't get time to think about the name. Did you do any research in the stdlib by any chance? I should have time to merge this tomorrow.

**Comment by AnirudhVyas** [comment]
Seems like you have handsful - My little one is in love with slides off late; so that kept me busy as well.  I'll take a look tomorrow - Thanks.

**Comment by magiconair** [comment]
There are examples for both `ToXXX()` and `Type()` in the stdlib. With the current set of methods I think `Map()` is a better fit. I'll make the change and merge it.

**Comment by magiconair** [comment]
I've merged the PR but apparently not in a way that auto-closes this one. :( Attribution is on you, though.

Thanks for the change and your quick replies.

**Pull request closed without merging**